### PR TITLE
payout proccess logs preparation - added placeholders

### LINKED ISSUE #1161: Add logs to payouts
Payouts have a complex process that might yield different results in each step, to be aware and control this process better we want to add a logging mechanism each step of the way.

### PR DESCRIPTION
**Pull request by noamso1**

### CODE REVIEW

**Comment by ariel-dao** [comment]
@noamso1 is there something I should review here at the moment?

**Comment by ariel-dao** [comment]
Looks great, some logs to add:
1. trustee decision log (approve or decline)
2. upload seller docs status

File: functions/src/payoutLogs/index.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import { RUNTIME_OPTIONS } from '../shared/configs/RUNTIME_OPTIONS';
+import { commonApp, commonRouter } from '../util';
+import { responseExecutor } from '../util/responseExecutor';
+
+import { createPayoutLog } from './business';
+
+const router = commonRouter();
+
+router.post('/create', async (req, res, next) => {
```
Comment (ariel-dao) [inline functions/src/payoutLogs/index.ts]: why do we need a route for this?

File: functions/src/payoutLogs/business/createPayoutLog.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { validate } from '../../util/validate';
+import { payoutLogDb } from '../database';
+import { IPayoutLog } from '../../shared/interfaces/IPayoutLog';
+import { CreatePayoutLogPayload } from '../../shared/types/CreatePayoutLogPayload';
+import { createPayoutLogDataValidationScheme } from '../../shared/schemas/createPayoutLogDataValidationScheme';
+
+/**
+ * Creates payoutLog for the provided payload. Please note that this is
```
Comment (ariel-dao) [inline functions/src/payoutLogs/business/createPayoutLog.ts]: what is not authenticated? and why?

File: functions/src/payments/triggers/payme/onPaymeSellerValidationApproved.ts [hunk 3]
```diff
@@ -47,18 +46,29 @@ export const onPaymeSellerValidationApproved = functions.firestore
...
+                createPayoutLog( { proposalId: proposal.id, description: 'Trying to transfer funds' } )
+
+                let transferFundsResponse = await transferFunds({
                     fromSeller: await getTrusteeStoreKey(),
                     toSeller: paymeSellerId,
                     actualAmountFromInvoices
                 })
 
-                const err = await withdrawBalance(proposerId) as IPaymeError;
+                if ( transferFundsResponse == true ) {
```
Comment (ariel-dao) [inline functions/src/payments/triggers/payme/onPaymeSellerValidationApproved.ts:57]: I've had a test that a transfer failed and it still went on to withdraw, please check

File: functions/src/notification/email/mailer.ts [hunk 4]
```diff
@@ -18,6 +18,11 @@ const setApiKey = async () => {
 export const sendMailDynamicTemplate = async (
   dynamicTemplateArgs: IDynamicTemplateArgs
 ) => {
+
+  if ( dynamicTemplateArgs.cbFunction ) {
+    dynamicTemplateArgs.cbFunction( 'try', dynamicTemplateArgs )
```
Comment (ariel-dao) [inline functions/src/notification/email/mailer.ts]: move [try, success, fail] to enums

Comment (noamso1) [reply functions/src/notification/email/mailer.ts]: Done

File: functions/src/payoutLogs/business/createPayoutLog.ts (hunk 2 above)
Comment (noamso1) [reply functions/src/payoutLogs/business/createPayoutLog.ts]: its just code i copied from elsewhere. irrelevant, deleted.

File: functions/src/payoutLogs/index.ts (hunk 1 above)
Comment (noamso1) [reply functions/src/payoutLogs/index.ts]: You're right, we don't !. removed.

File: functions/src/payments/triggers/payme/onPaymeSellerValidationApproved.ts (hunk 3 above)
Comment (noamso1) [reply functions/src/payments/triggers/payme/onPaymeSellerValidationApproved.ts:57]: Yes. if axios returns an error, the TransferFunds function returned without an error. FIXED.

**Review by ariel-dao — APPROVED** [review]

**Pull request merged**